doc/SONiC_QoS_Mgmt_HLD-1.0.md

```diff
@@ -0,0 +1,499 @@
+    openconfig-qos-interfaces.yang:: qos-interfaces/interface/interface-id/output/queues/queue/state  
+    Augment Open Config YANG model with new definitions for queue statistics and watermark statistics.
+
+     
+- Threshold  
+    Augment Open Config YANG model with new definitions of Queue Thresholds.
+
+
+
+### 3.2.2 CLI
+
+#### 3.2.2.1 Configuration Commands
+
+- Configuring WRED
+````
+sonic(config)# wred <name>
+sonic(conf-wred-<name>)# random-detect color [green|yellow|red] minimum-threshold 100 maximum-threshold 300
+drop-probability 40
+sonic(conf-wred-<name>)# random-detect ecn
+
+    sonic(config)# no wred <name>
+````
+
+- Configure Scheduler Policy
+````
+  sonic(config)# scheduler-policy <name>
+  sonic(conf-sched-policy)# scheduler <q#> 
+  sonic(conf-sched-policy-q)# priority level <val>
+  sonic(conf-sched-policy-q)# shape-bandwidth min <val> min-burst <val> max <val> max-burst <val> 
+  sonic(conf-sched-policy-q)# bandwidth-percent <val>
+
+  sonic(conf-sched-policy)# no scheduler <q#>
+  sonic(config)# no scheduler-policy <name>   
+````
+
+
+- Config DSCP to Traffic Class map  
+````
+  sonic(config)# qos-map dscp-to-tc-map <name>
+  sonic(conf-qos-map)# traffic-class <1..8> dscp  {<dscp-value>}
+
+  sonic(config)# no dscp-to-tc-map <name>
+````
+
+- Config Traffic Class to Queue map
+````
+  sonic(config)# qos-map tc-to-queue-map <name>
+  sonic(conf-qos-map)# queue 3 traffic-class 0-3
+````
+
+- Config interface QoS
+````
+  sonic(config)# interface <name>
+  sonic(conf-if-name)# dscp-to-tc-map <name>
+  sonic(conf-if-name)# tc-to-queue-map <name>
+  sonic(conf-if-name)# scheduler-policy <name>
+  sonic(conf-if-name)# queue <q#> wred-profile <wred-profile-name>
+  sonic(conf-if-name)# queue <q#> [ucast|mcast] threshold <val> 
+     
+  sonic(conf-if-name)# no dscp-to-tc-map <name>
+  sonic(conf-if-name)# no tc-to-queue-map <name>
+  sonic(conf-if-name)# no scheduler-policy <name>
+  sonic(conf-if-name)# no queue <q#> wred-profile <wred-profile-name>
+  sonic(conf-if-name)# no queue <q#> [ucast|mcast] threshold 
+
+````
+- Config Watermark polling interval
+````
+  sonic(config)# qos watermark polling-interval <val>
+````
+
+
+#### 3.2.2.2 Show Commands
+
+- Show WRED
+````
+  >show wred {<name>}
+  Sample Output:
+     > show wred wred_1
+    Profile Name | Green            | Yellow           | Red              |       |    
+                 |------------------|------------------|------------------|       |    
+                 | MIN MAX DROP-RATE| MIN MAX DROP-RATE| MIN MAX DROP-RATE|  ECN  |
+                 | KB  KB  %        | KB  KB  %        | KB  KB  %        |       |  
+    -------------|------------------|------------------|------------------|-------|
+      wred_1     | 100 1000 100     | 50 100 100       | 50 100 100       | All   |
+    -------------|------------------|------------------|------------------|-------|
+````
+
+
+- Show Scheduler Policy  
+````    
+  >show scheduler-policy <name>   

```

Comment:
Can we make the o/p more structured? Easier to read and reduces the number of lines too e.g.
Schedule      Type          Bandwidth      Priority   Min         Min-Burst        Max        Max-Burst
q0                 Priority      NA                 1             100          16kBytes         200         16kBytes
q1                 WRR      40                  NA           200          16kBytes         1000        16kBytes